src/search-page-controller.js

```diff
@@ -154,58 +156,188 @@ class SearchPageController {
+                        // Render charts
+                        //
+                        const chart = new DatasetChart();
+                        const container = d3.select('#google-charts-container');
+
+                        data.forEach((datum, index) => {
+
+                            // Replace the entity ids in the data with the region names
+                            //
+                            for (var i = 1; i < datum.data[0].length; i++) {
+                                var dataItem = datum.data[0][i];
+                                this.params.regions.forEach(region => {
+                                    if (dataItem == region.id)
+                                        datum.data[0][i] = region.name;
+                                });
+                            }
+
+                            // Render the chart
+                            //
+                            const params = dataValueParams[index];
+                            chart.render(dataset.id, params.chartId, datum.data);
+                        });
+
+                    }, error => console.error(error));
+
+                    MapView.create(this.params.regions, variable, {year: this.params.year}, this.params)
+                        .then(map => map.show('div#map'))
+                        .catch(error => {
+                            throw error;
+                        });
+
+                }, error => console.error(error));
+
+            }, error => console.error(error));
         }
 
+        // Map summary links
+        //
         $('.map-summary-more').click(() => {
             $('.map-summary-links').toggle();
             $('.map-summary-more').text($('.map-summary-more').text() == 'More Information' ? 'Less Information' : 'More Information');
         });
 
-        this.subMenus();
-
         // Map should display above charts on the desktop, below charts on mobile.
         //
-        $(window).resize(() => this.moveMap());
-        this.moveMap();
+        $(window).resize(() => this.moveMapNode());
+        this.moveMapNode();
+    }
+
+    getContraintByValueOrDefault(permutations, year) {
+
+        const constraintValue = year.toLowerCase();
+
+        for (var i = 0; i < permutations.length; i++) {
+
+            var constraint = permutations[i];
+
+            if (constraint.constraint_value == constraintValue)
+                return constraint;
+        }
+
+        return permutations[permutations.length - 1];
+    }
+
+    getVariableByNameOrDefault(variablesArray, metric) {
+
+        // If metric is empty, use first variable
+        //
+        if (metric.length == 0)
+            return variablesArray[0];
+
+        // Find variable by name using the metic
+        //
+        const variableName = metric.replace(/_/g, ' ').toLowerCase();
+
+        for (var i = 0; i < variablesArray.length; i++) {
+
+            var variable = variablesArray[i];
+
+            if (variable.name == variableName)
```

Comment:
We should use `variable.id` instead of `variable.name` since names may change.

Also, using lodash you could simply do

```js
getVariable(variables, id) {
    return _.find(variables, {id}) || variables[0];
}
```